felix/wireguard/route_table.go

```diff
@@ -0,0 +1,24 @@
+package wireguard
+
+import (
+	"github.com/projectcalico/calico/felix/ifacemonitor"
+	"github.com/projectcalico/calico/felix/ip"
+	"github.com/projectcalico/calico/felix/routetable"
+)
+
+// routeTableSyncer is the interface used to manage data-sync of route table managers. This includes notification of
+// interface state changes, hooks to queue a full resync and apply routing updates.
+type routeTableSyncer interface {
+	OnIfaceStateChanged(string, ifacemonitor.State)
+	QueueResync()
+	Apply() error
+}
+
+// routeTable is the interface provided by the standard routetable module used to progam the RIB.
+type routeTable interface {
+	routeTableSyncer
+	SetRoutes(ifaceName string, targets []routetable.Target)
+	SetL2Routes(ifaceName string, targets []routetable.L2Target)
+	RouteRemove(ifaceName string, cidr ip.CIDR)
+	RouteUpdate(ifaceName string, target routetable.Target)
+}
```

Comment:
I'm not sure why you need to add this file.

Reply:
So, Wireguard was using the `RouteTable` struct directly from the `routetable` package, and that meant that I could not override it directly with the `DummyTable`. All other uses of `RouteTable` went through the `routeTable` interface, from `intdataplane` package.

That interface did not include the `RouteRemove` and `RouteUpdate` methods `Wireguard` uses; and I didn't want to introduce a dependency between `wireguard` and `intdataplane` packages for this; so I replicated the interface with just the methods `wireguard` needs, checking that both `routetable.RouteTable` and `DummyTable` implements them.

That was my logic, but of course I could be missing something, do you have any suggestions on how would an alternative look like?